csv.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+type csvPrinter struct {
+	writer     *csv.Writer
+	file       *os.File
+	filename   string
+	headerDone bool
+}
+
+const (
+	csvTimeFormat = "2006-01-02 15:04:05.000"
+)
+
+func newCSVPrinter(filename string, args []string) (*csvPrinter, error) {
+	file, err := os.Create(filename)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CSV file: %w", err)
+	}
+
+	writer := csv.NewWriter(file)
+	return &csvPrinter{
+		writer:     writer,
+		file:       file,
+		filename:   filename,
+		headerDone: false,
+	}, nil
+}
+
+func (cp *csvPrinter) writeHeader() error {
+	header := []string{
+		"Event Type", "Timestamp", "Address", "Hostname", "Port",
+		"Hostname Resolve Retries", "Total Successful Probes", "Total Unsuccessful Probes",
+		"Never Succeed Probe", "Never Failed Probe", "Last Successful Probe",
+		"Last Unsuccessful Probe", "Total Packets", "Total Packet Loss",
+		"Total Uptime", "Total Downtime", "Longest Uptime", "Longest Uptime Start",
+		"Longest Uptime End", "Longest Downtime", "Longest Downtime Start",
+		"Longest Downtime End", "Latency Min", "Latency Avg", "Latency Max",
+		"Start Time", "End Time", "Total Duration",
+	}
+	return cp.writer.Write(header)
+}
+
+func (cp *csvPrinter) saveStats(tcping tcping) error {
+	if !cp.headerDone {
+		if err := cp.writeHeader(); err != nil {
+			return fmt.Errorf("error writing CSV header: %w", err)
+		}
+		cp.headerDone = true
+	}
+
+	totalPackets := tcping.totalSuccessfulProbes + tcping.totalUnsuccessfulProbes
+	packetLoss := float64(tcping.totalUnsuccessfulProbes) / float64(totalPackets) * 100
+
+	lastSuccessfulProbe := tcping.lastSuccessfulProbe.Format(csvTimeFormat)
+	lastUnsuccessfulProbe := tcping.lastUnsuccessfulProbe.Format(csvTimeFormat)
+	if tcping.lastSuccessfulProbe.IsZero() {
+		lastSuccessfulProbe = ""
+	}
+	if tcping.lastUnsuccessfulProbe.IsZero() {
+		lastUnsuccessfulProbe = ""
+	}
+
+	longestUptimeDuration := tcping.longestUptime.duration.String()
+	longestUptimeStart := tcping.longestUptime.start.Format(csvTimeFormat)
+	longestUptimeEnd := tcping.longestUptime.end.Format(csvTimeFormat)
+	longestDowntimeDuration := tcping.longestDowntime.duration.String()
+	longestDowntimeStart := tcping.longestDowntime.start.Format(csvTimeFormat)
+	longestDowntimeEnd := tcping.longestDowntime.end.Format(csvTimeFormat)
```

Comment:
These show a funky result when there are no failures.

![Screenshot from 2024-11-06 10-38-50](https://github.com/user-attachments/assets/344fe88a-c0c5-4066-81d3-09887d3145b9)


You can follow what statsPrinter does to fix it.

Reply:
btw, was this addressed?

Reply:
Hmm I think this was older code inspired by our db printer, The same fields don't exist for our csv printer and as for  any weird results in the equivalent fields in our csv printer when facing similar scenarios, I don't think I saw any...

Reply:
something looks funky with the dates. I will fix it and ping you....Thanks for the tip @pouriyajamshidi

Reply:
alright, thanks.

Reply:
Hey @pouriyajamshidi could you please take a look whenever you find time?

Reply:
Hey @SYSHIL 

Sure!